### PR TITLE
feat(P-052): build system additions and linked investment settings flows

### PR DESCRIPTION
**Pull request by johnsherlock**
## Summary

- Replaces the old flat finance form (`/settings/finance`) with a multi-record system additions list, showing empty, single, and multi-record states with per-record derived totals (total investment, remaining months, outstanding balance)
- Implements add (`/settings/finance/new`) and edit/delete (`/settings/finance/[id]/edit`) flows with validation matching the domain rules from P-051
- Removes all dead code tied to the old `installations.finance_*` flat columns: `FinanceForm.tsx`, `finance-domain.ts`, `finance-types.ts`, `finance-domain.test.ts`, `loadFinanceSettings()`, and the old `saveFinanceSettings` action
- Adds `drizzle/0013_drop_flat_finance_columns.sql` migration to drop the 5 flat finance columns
- Cleans up `range/loader.ts`, `live/loader.ts`, `RangeHistoryScreen.tsx`, and `range/page.tsx` which read those columns (removes the old finance repayment panel)
- Updates `seed.ts` to replace the flat finance seed data with a `system_additions` record

## Test plan

- [ ] `/settings/finance` shows empty state for a new installation
- [ ] "Add system addition" navigates to the form, saves a record, and returns to the list
- [ ] List shows derived totals (total investment, remaining months, outstanding balance)
- [ ] "Edit" link on a card loads the form pre-populated with existing values
- [ ] Saving edits updates the record and returns to the list
- [ ] "Delete this addition" on the edit page removes the record and returns to the list (including last-record → empty state)
- [ ] Validation rejects: missing label, missing date, no payment fields, monthly repayment without duration
- [ ] Settings home `/settings` shows Finance as complete when at least one valid record exists, incomplete when zero records remain
- [ ] All 389 vitest tests pass; zero TypeScript errors

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by johnsherlock** [comment]
Review against `P-052` / `FE-014` on head `d5dc37d376b35714309cbbed94001082fa164e33`:

1. `apps/web/app/range/RangeHistoryScreen.tsx`
`P-052` / `FE-014` say system additions should unlock the first downstream finance behaviour, and the new Finance settings copy now explicitly promises a payback progress tracker. This PR removes `PaybackTracker` from Range History without adding a replacement, so completing the new setup flow no longer unlocks any finance insight surface at all. Either keep the existing tracker until `U-051` / `P-049` lands, or ship a replacement surface in this story so the product promise remains true.

2. `apps/web/src/finance/system-addition-domain.ts`
The new validator only checks whether payment fields are non-null, so a record with `upfrontPayment = 0` or `monthlyRepayment = 0` plus duration is accepted and then counted as a configured Finance setup. That conflicts with `FE-014`, which says users who do not want to share investment details should not complete this domain yet, and it lets meaningless records advance setup progress. Tighten validation to require at least one positive payment amount and add regression tests for zero-value cases.

File: apps/web/app/range/RangeHistoryScreen.tsx [hunk 1]
```diff
@@ -317,16 +312,6 @@ export function RangeHistoryScreen({ payload, today, financeMode, monthlyFinance
                   currency={payload?.meta.currency ?? 'EUR'}
```
Comment (johnsherlock) [inline apps/web/app/range/RangeHistoryScreen.tsx:313]: `P-052` / `FE-014` say system additions should unlock the first downstream finance behaviour, and the new Finance settings copy now explicitly promises a payback progress tracker. Removing this block without adding a replacement means completing the new setup flow no longer unlocks any finance insight surface at all. Either keep the existing tracker until `U-051` / `P-049` lands, or ship a replacement surface in this story so the product promise remains true.

File: apps/web/app/settings/finance/actions.ts [hunk 2]
```diff
@@ -26,47 +28,59 @@ type SchemaModule = typeof import('@/src/db/schema');
...
+    (!Number.isInteger(repaymentDurationMonths) || repaymentDurationMonths < 1)
   ) {
-    return { ok: false, error: 'Invalid numeric input.' };
+    return { ok: false, error: 'Repayment duration must be a whole number of months.' };
+  }
+  if (capacityAddedKw !== null && !Number.isFinite(capacityAddedKw)) {
+    return { ok: false, error: 'Invalid capacity value.' };
   }
 
-  const validation = validateFinanceInputs({
```
Comment (johnsherlock) [inline apps/web/app/settings/finance/actions.ts:68]: This new save flow now relies on `validateSystemAdditionInputs`, which only rejects `null` payment fields. That means `upfrontPayment = 0` or `monthlyRepayment = 0` plus duration will be accepted here and persisted as if the user had supplied meaningful finance data. In `FE-014`, users who do not want to share investment details are supposed to leave this domain incomplete, so I think this path should require at least one positive payment amount and add regression coverage for zero-value cases.

**Review by johnsherlock — COMMENTED** [review]
Inline follow-up review against `P-052` / `FE-014` on head `d5dc37d376b35714309cbbed94001082fa164e33`, using diff-position anchors.
